security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/SecurityAnalysisExecutionHandler.java

```diff
@@ -0,0 +1,186 @@
+                                            int executionCount,
+                                            SecurityAnalysisExecutionInput input) {
+        this.reader = reader;
+        this.optionsCustomizer = optionsCustomizer;
+        this.executionCount = executionCount;
+        this.input = requireNonNull(input);
+    }
+
+    /**
+     * Copies case file, contingencies file, and parameters file to working directory,
+     * and creates the {@literal itools security-analysis} command(s) to be executed,
+     * based on configuration and the optional options customizer.
+     */
+    @Override
+    public List<CommandExecution> before(Path workingDir) throws IOException {
+        CommandExecution execution = createSecurityAnalysisCommandExecution(workingDir);
+        return Collections.singletonList(execution);
+    }
+
+    /**
+     * Reads result from the working directory, as defined by the specified reader.
+     */
+    @Override
+    public R after(Path workingDir, ExecutionReport report) throws IOException {
+        super.after(workingDir, report);
+        LOGGER.debug("End of command execution in {}. ", workingDir);
+        return reader.read(workingDir);
+    }
+
+    /**
+     * Create the {@literal itools security-analysis} command and copies necessary files to working directory.
+     * Options may be added through the specified {@link #optionsCustomizer}
+     */
+    private CommandExecution createSecurityAnalysisCommandExecution(Path workingDir) {
+        SecurityAnalysisCommandOptions options = new SecurityAnalysisCommandOptions()
+                .id("security-analysis")
+                .resultExtensions(input.getResultExtensions())
+                .violationTypes(input.getViolationTypes());
+
+        addCaseFile(options, workingDir, input.getNetworkVariant());
+        addParametersFile(options, workingDir, input.getParameters());
+        input.getContingenciesSource().ifPresent(
+            source -> addContingenciesFile(options, workingDir, source)
+        );
+
+        if (executionCount > 1) {
+            options.task(taskNumber -> new Partition(taskNumber + 1, executionCount));
+        }
+
+        if (optionsCustomizer != null) {
+            optionsCustomizer.customizeOptions(workingDir, options);
+        }
+
+        return new CommandExecution(options.toCommand(), executionCount);
+    }
+
+    private static Path getCasePath(Path workingDir) {
+        return workingDir.resolve(NETWORK_FILE);
+    }
+
+    private static Path getParametersPath(Path workingDir) {
+        return workingDir.resolve(PARAMETERS_FILE);
+    }
+
+    private static Path getContingenciesPath(Path workingDir) {
+        return workingDir.resolve(CONTINGENCIES_FILE);
+    }
+
+    /**
+     * Add case file option, and write network to working directory.
+     */
+    private void addCaseFile(SecurityAnalysisCommandOptions options, Path workingDir, NetworkVariant variant) {
+        Path dest = getCasePath(workingDir);
```

Comment:
It seems that the folder should exist: who is in charge to create it? Same remark for the other folders

Reply:
ComputationManager provide working directory to handler.